delta/plugins/graph-analytics/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/graph/analytics/model/JsonLdDocument.scala

```diff
@@ -41,37 +42,34 @@ object JsonLdDocument {
     }
   }
 
-  val empty: JsonLdDocument = JsonLdDocument(Set.empty, Set.empty, Set.empty)
+  val empty: JsonLdDocument = JsonLdDocument(Set.empty, Set.empty, Set.empty, Set.empty)
 
-  def singleEntry(entry: JsonLdEntry): JsonLdDocument = JsonLdDocument(Set(entry), Set.empty, Set.empty)
+  def singleEntry(entry: JsonLdEntry): JsonLdDocument = JsonLdDocument(Set(entry), Set.empty, Set.empty, Set.empty)
 
   /**
     * Creates a [[JsonLdDocument]] from a expanded JSON-LD
     * @param expanded
     *   the expanded JSON-LD to analyze
-    * @param findRelationship
+    * @param findRelationships
     *   the function to apply to resolve resources/files
     */
   def fromExpanded(
       expanded: ExpandedJsonLd,
-      findRelationship: Iri => UIO[Option[Set[Iri]]]
+      findRelationships: Set[Iri] => UIO[Map[Iri, Set[Iri]]]
```

Comment:
What changes is that we now attempt to resolve all references in the end instead of one by one
Otherwise the code is the same